lib/gems/pending/util/mount/miq_swift_session.rb

```diff
@@ -0,0 +1,120 @@
+        disconnect
+        logger.error("Error creating Swift container #{@container_name}. #{err}")
+        msg = "Error creating Swift container #{@container_name}. #{err}"
+        raise err, msg, err.backtrace
+      end
+    rescue => err
+      disconnect
+      logger.error("Error getting Swift container #{@container_name}. #{err}")
+      msg = "Error getting Swift container #{@container_name}. #{err}"
+      raise err, msg, err.backtrace
+    end
+    #
+    # Get the remote path, and parse out the bucket name.
+    #
+    object_file_with_bucket = URI.split(URI.encode(uri))[5]
+    object_file = object_file_with_bucket.split(File::Separator)[2..- 1].join(File::Separator)
+    #
+    # write dump file to swift
+    #
+    logger.debug("Writing [#{local_file}] to Bucket [#{@container_name}] using object file name [#{object_file}]")
+    begin
+      container.files.create(:key => object_file, :body => File.open(local_file))
+    rescue Excon::Errors::Unauthorized => err
+      disconnect
+      logger.error("Access to Swift container #{@container_name} failed due to a bad username or password. #{err}")
+      msg = "Access to Swift container #{@container_name} failed due to a bad username or password. #{err}"
+      raise err, msg, err.backtrace
+    rescue => err
+      disconnect
+      logger.error("Error uploading #{local_file} to Swift container #{@container_name}. #{err}")
+      msg = "Error uploading #{local_file} to Swift container #{@container_name}. #{err}"
+      raise err, msg, err.backtrace
+    end
+  end
+
+  private
+
+  def swift
+    require 'manageiq/providers/openstack/legacy/openstack_handle'
+    extra_options = {
+      :ssl_ca_file    => ::Settings.ssl.ssl_ca_file,
+      :ssl_ca_path    => ::Settings.ssl.ssl_ca_path,
+      :ssl_cert_store => OpenSSL::X509::Store.new
+    }
+    extra_options[:domain_id] = @domain_id
+    extra_options[:omit_default_port] = ::Settings.ems.ems_openstack.excon.omit_default_port
+    extra_options[:read_timeout]      = ::Settings.ems.ems_openstack.excon.read_timeout
+    extra_options[:service] = "Compute"
+
+    @osh ||= OpenstackHandle::Handle.new(@username, @password, @host, @port, @api_version, @security_protocol, extra_options)
+    @osh.connection_options = {:instrumentor => $fog_log}
+    begin
+      @swift ||= @osh.swift_service
+    rescue Excon::Errors::Unauthorized => err
+      disconnect
+      logger.error("Access to Swift host #{@host} failed due to a bad username or password. #{err}")
+      msg = "Access to Swift host #{@host} failed due to a bad username or password. #{err}"
+      raise err, msg, err.backtrace
+    rescue => err
+      disconnect
+      logger.error("Error connecting to Swift host #{@host}. #{err}")
+      msg = "Error connecting to Swift host #{@host}. #{err}"
+      raise err, msg, err.backtrace
+    end
+  end
+
+  def query_params(query_string)
+    query_string.split('&').each do |query|
+      query_parts = query.split('=')
```

Comment:
I think this would be easier to read as `key, value = query.split('=')`

Reply:
Sure no problem.